dot/parachain/availability-store/availability_store.go

```diff
@@ -459,32 +481,247 @@ func (av *AvailabilityStoreSubsystem) processMessages() {
 				if err != nil {
 					logger.Errorf("failed to process active leaves update signal: %w", err)
 				}
+
 			case parachaintypes.BlockFinalizedSignal:
-				av.ProcessBlockFinalizedSignal(msg)
+				err := av.ProcessBlockFinalizedSignal(msg)
+				if err != nil {
+					logger.Errorf("failed to process block finalized signal: %w", err)
+				}
 
 			default:
-				logger.Error(parachaintypes.ErrUnknownOverseerMessage.Error())
+				if msg != nil {
+					logger.Infof("unknown message type %T", msg)
+					logger.Error(parachaintypes.ErrUnknownOverseerMessage.Error())
+					// this error shouldn't happen, so we'll panic to catch it during development
+					panic(parachaintypes.ErrUnknownOverseerMessage.Error())
```

Comment:
Info, error log, and panic will print the same thing. Panic is enough. 

```suggestion
panic(fmt.Sprintf("%s: %T", parachaintypes.ErrUnknownOverseerMessage.Error(), msg))
```

Reply:
updated.